app/src/main/java/com/breadwallet/tools/security/BRKeyStore.java

```diff
@@ -228,6 +229,27 @@ private static SecretKey createKeys(String alias, boolean auth_required) throws
     private synchronized static byte[] _getData(final Context context, String alias, String alias_file, String alias_iv, int request_code)
             throws UserNotAuthenticatedException {
         validateGet(alias, alias_file, alias_iv);//validate entries
+
+        try {
+            lock.lock();
+            return BreadApp.keyStoreManager.getDataBlocking(new AliasObject(alias, alias_file, alias_iv));
+        } catch (UserNotAuthenticatedException e) {
+            Timber.e(e, "timber:_getData: showAuthenticationScreen: %s", alias);
+            showAuthenticationScreen(context, request_code, alias);
```

Comment:
https://developer.android.com/privacy-and-security/keystore#UserAuthentication

Reply:
I have questions @kcw-grunt, our Android app will still need user's pin lock screen?

Reply:
still thinking about this how to migrate without the device lock, since the key (for phrase and canary) that created before need to do device unlock first. related with link above

ref:
- https://github.com/litecoin-foundation/litewallet-android/blob/7b5a20d87aba277833abe7c3245732a074279020/app/src/main/java/com/breadwallet/tools/security/BRKeyStore.java#L220
- https://github.com/litecoin-foundation/litewallet-android/blob/7b5a20d87aba277833abe7c3245732a074279020/app/src/main/java/com/breadwallet/tools/security/BRKeyStore.java#L156
- https://github.com/litecoin-foundation/litewallet-android/blob/7b5a20d87aba277833abe7c3245732a074279020/app/src/main/java/com/breadwallet/tools/security/BRKeyStore.java#L384
- https://github.com/litecoin-foundation/litewallet-android/blob/7b5a20d87aba277833abe7c3245732a074279020/app/src/main/java/com/breadwallet/tools/security/BRKeyStore.java#L409